### ISSUE TITLE
Accessibility Issues for the Test Request Dialog / API Client

### ISSUE THREAD

**Issue by kayla-beckham** [scalar/scalar#3441]
### What happens?

This is the biggest section.
![Screen Shot 2024-10-10 at 9 29 57 AM](https://github.com/user-attachments/assets/1902c556-123e-4b8f-af85-dddfc9a998db)

1. Test request pane should have role=‘dialog’ and an aria label with an accessible name. Both the role and the name should be announced by the screen reader.
2. When the test request dialog is opened, keyboard focus stays on the page. Focus should move to the dialog when it is opened.
    1. Moving the focus past all of the active elements causes the focus to leave the dialog. The focus should remain in the dialog until it is closed.
3. The url edit field for not have an accessible name to be announced by the screen reader. It needs an aria label.
5. In Test request dialog, when the request accordion sections for Authentication, Cookies, etc are expanded, the screenreader does not announce the accordion header when the focus is moved to the fields inside of the accordion. Need to add role=‘region’ to the expanded area
    1.  there is not a visible focus indicator border on the text input fields in the Authentication/Cookies/Headers input sections.
    2. The clear button is nested inside of the accordion and announced twice by the screen reader. It inherits the expanded/collapsed state. It should be separated from accordion element
    3. The checkboxes beside the key/value inputs are not keyboard operable and they do not have names. They should have labels that associate them with the second column of the table.
    4. Accordion label should be implemented as heading <h#> element
![Screen Shot 2024-10-10 at 9 30 15 AM](https://github.com/user-attachments/assets/3d92ac9f-5a94-42f5-aa50-1bc9b1176354)
6. In the test request dialog, the menu to select what inputs should be shown in the request pane is not visible when navigated by keyboard (it shows up when mouse overs over the area)
    1. This menu is implemented as buttons, but acts as a set of tabs. The screenreader should announce each item in the menu like ‘All tab 1 of 6’
![Screen Shot 2024-10-10 at 9 31 47 AM](https://github.com/user-attachments/assets/e4d74dbe-3394-4844-9216-921a00d5a431)
7. In the test request dialog, the Auth Type combo box does not have an accessible name announced by the screen reader (aria-label)
    1. Needs role=checkbox. 
    3. The checkbox is skipped during keyboard navigation. Pressing tab on the combobox should go to the checkbox, not the token input
    4. The chip that indicates the combobox selection is not keyboard accessible. Mouse users can cancel the chip without opening the combobox, but keyboard users cannot access the chip.
        1. The chip should have aria-label and aria-labelled-by attributes that indicate the action (remove) and the visible label (the authentication type)
8. In the table that makes up the key/value pairs in of Authentication in the request pane of the test request dialog, the table header <th> tag is set as the first row of the table, but the headers are actually defined in the columns of the table.
    1. The other headers in the request and response panes are all missing headers. “Key” and “Value” should be defined as the headers of the tables.
![Screen Shot 2024-10-10 at 9 34 36 AM](https://github.com/user-attachments/assets/e4d07351-7c47-4c9d-8968-2457b9fb9e45)
9. When variables are defined in the request pane, the information button does not have an accessible name. It needs an aria-label to be announced by the screen reader.
10. When there is a code editor in the Body field within the test request dialog, the keyboard focus gets trapped inside of there. Users should be able to move keyboard focus out of the editor.
11. When a request is sent in the Test Request dialog, the screen reader should announce a status message that a response is received.
12. The request history button is missing an aria-label
13. Scrollable code snippet in the response pane is not accessible to the keyboard.
14. The button to expand/collapse collections side panel in the test request dialog is missing the aria-label and aria-expanded: true/false attributes
15. When the collections side menu is opened, the tab focus moves next to the url field. After the user opens the side panel with the keyboard, tab focus should proceed to the search field in the side panel.
16. Links in the side panel receive keyboard focus twice.
17. When the collections side panel in the test request pane is used, activating a link in the side panel should move keyboard focus to the url field. Currently the focus stays in the collection side panel
18. The close button for the Test Request dialog is missing an aria-label

### What did you expect to happen?

The test request dialog should be announced by a screen reader when it is opened, and focus should move into the dialog and stay there until the dialog is closed.
Active elements should be keyboard accessible, have visual indicators of keyboard focus, and have meaningful, contextual aria-labels so they can be announced by screen readers.
The tables and other nested elements for defining parameters should be defined with correct hierarchies and aria attributes. The <th> elements should be correctly attributed to the fields of the table that act as headings.
The code editor for request bodies should not trap keyboard focus. It should be possible to move the keyboard focus in and out of the editor.
The code in the response code snippet should be keyboard accessible and it should be announced by a screen reader.
Sending a request should result in a status announcement from the screen reader.
Instructions in the footer of the Test Request dialog should be announced when they appear (Send Request command enter)
When the side menu is opened, focus should move into the menu, and every item in the menu should receive keyboard focus only once.
The button for opening the side menu should announce an open/close status.
The button to close the test request dialog should have an aria label so it can be announced.
